addons/sale_loyalty/models/sale_order.py

```diff
@@ -291,6 +296,13 @@ def _discountable_order(self, reward):
         base_lines = []
         for line in order_lines:
             base_line = line._prepare_base_line_for_taxes_computation()
+            taxes = base_line['tax_ids'].flatten_taxes_hierarchy()
+            if not reward.program_id.is_payment_program:
+                # To compute the discountable amount we get the subtotal and add
+                # non-fixed tax totals. This way fixed taxes will not be discounted
+                # This does not apply to Gift Cards and e-Wallet, where the total
+                # order amount may be paid with the card balance
+                taxes = taxes.filtered(lambda t: t.amount_type != 'fixed')
             base_line['discount_taxes'] = base_line['tax_ids'].flatten_taxes_hierarchy().filtered(lambda tax: tax.amount_type != 'fixed')
```

Comment:
???
This var will not be used, and will be overwritten in L326. Additionally, doesn't L306 do something similar anyway?

Reply:
Thanks for the note, I forgot to change the next line :smile: